harness/determined/keras/callbacks.py

```diff
@@ -644,3 +647,54 @@ def get_state(self) -> Any:
 
     def load_state(self, state: Any) -> None:
         _tf_keras_callback_load_state(self, state)
+
+
+class TensorBoard(tf.keras.callbacks.TensorBoard, Callback):  # type: ignore
+    """
+    This is a thin wrapper over the TensorBoard callback that ships with ``tf.keras``.  For more
+    information, see the :ref:`TensorBoard Guide <how-to-tensorboard>` or the upstream docs for
+    `tf.keras.callbacks.TensorBoard
+    <https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/TensorBoard>`__.
+
+    Note that if a ``log_dir`` argument is passed to the constructor, it will be ignored.
+    """
+
+    def __init__(self, *args: Any, **kwargs: Any):
+        self.workload_end_count = 0
+        user_log_dir = kwargs.pop("log_dir", None)
+        if user_log_dir is not None:
+            logging.warning(
+                f"arg log_dir={user_log_dir} to det.keras.callbacks.TensorBoard will be ignored"
+            )
+        log_dir = str(tensorboard.get_base_path({}).resolve())
+        tf.keras.callbacks.TensorBoard.__init__(self, log_dir=log_dir, *args, **kwargs)
+
+    def _write_logs(self, *args: Any) -> None:
+        """
+        _write_logs calls the original _write_logs() function from the Keras
+        TensorBoard callback. After the logs are flushed to disk, we close and
+        reopen the tf event writer so that it serializes the next set of logs
+        to a new file. This allows the tensorboard manager to treat the
+        written files as immutable and upload them to persistent storage
+        without later having to append to them. This behavior is useful for
+        tensorboard backed by S3.
+        """
+        tf.keras.callbacks.TensorBoard._write_logs(self, *args)
+        self.writer.close()
+        self.writer.reopen()
+
+    def get_state(self) -> Any:
+        return {"workload_end_count": self.workload_end_count}
+
+    def load_state(self, state: Any) -> None:
+        self.self.workload_end_count = state["workload_end_count"]
+
+    # Translate train workload hooks into epoch hooks.
+    def on_train_workload_begin(
+        self, total_batches_trained: int, batches_requested: Optional[int], logs: Dict
+    ) -> None:
+        tf.keras.callbacks.TensorBoard.on_epoch_begin(self, self.workload_end_count, logs)
+
+    def on_train_workload_end(self, total_batches_trained: int, logs: Dict) -> None:
+        tf.keras.callbacks.TensorBoard.on_epoch_end(self, self.workload_end_count, logs)
```

Comment:
question: aren't these already get logged in Determined automatically here: https://github.com/determined-ai/determined/blob/b1c8c1cbab0276a419cb4d51a4e72e5de2844957/harness/determined/tensorboard/metric_writers/callback.py#L43

Reply:
I mean yes, but my goal here is to match the behavior that we had before.  This is what things looked like for me on tip of master:
![tb-master](https://user-images.githubusercontent.com/47791514/98995221-426a2100-24ee-11eb-8ba1-937f54016fa7.png)

And this is what the same experiment looked like in 0.13.6:
![tb-0 13 6](https://user-images.githubusercontent.com/47791514/98995261-5150d380-24ee-11eb-8a74-ca31a0110395.png)

Reply:
Interesting, not sure why we are seeing this behavior but we definitely want the latter.